runtime_api.go

```diff
@@ -0,0 +1,183 @@
+		// 2. An additional case statement that handles printing runtime.hex
+		//    values. Without this case statement, the default case will print
+		//    the runtime.hex values in a way not consistent with normal panic
+		//    outputs
+		for _, decl := range file.Decls {
+			funcDecl, ok := decl.(*ast.FuncDecl)
+			if ok && funcDecl.Name.Name == "printany" {
+				for _, stmt := range funcDecl.Body.List {
+					switchStmt, ok := stmt.(*ast.TypeSwitchStmt)
+					if !ok {
+						continue
+					}
+
+					switchStmt.Body.List = append(switchStmt.Body.List, printanyHexCase)
+					break
+				}
+
+				funcDecl.Body.List = append([]ast.Stmt{hidePanicsCheckStmt}, funcDecl.Body.List...)
+				break
+			}
+		}
+
+		file.Decls = append(file.Decls, panicprintDecl)
+	case "panic.go", "traceback.go":
+		// Change all calls to print, which we don't control, to
+		// panicprint, which we do control and does the same thing.
+		ast.Inspect(file, switchPanicPrints)
+	}
+}
+
+var hidePanicsCheckStmt = &ast.IfStmt{
+	Cond: &ast.Ident{Name: "hidePanics"},
+	Body: &ast.BlockStmt{List: []ast.Stmt{
+		&ast.ReturnStmt{},
+	}},
+}
+
+var hidePanicsDecls = []ast.Decl{
+	&ast.GenDecl{
+		Tok: token.VAR,
+		Specs: []ast.Spec{
+			&ast.ValueSpec{
+				Names: []*ast.Ident{
+					&ast.Ident{Name: "hidePanics"},
+				},
+				Type: &ast.Ident{Name: "bool"},
+			},
+		},
+	},
+	&ast.FuncDecl{
+		Name: &ast.Ident{Name: "HidePanics"},
+		Type: &ast.FuncType{
+			Params: &ast.FieldList{
+				List: []*ast.Field{
+					&ast.Field{
+						Names: []*ast.Ident{
+							&ast.Ident{Name: "hide"},
+						},
+						Type: &ast.Ident{Name: "bool"},
+					},
+				},
+			},
+		},
+		Body: &ast.BlockStmt{
+			List: []ast.Stmt{
+				&ast.AssignStmt{
+					Lhs: []ast.Expr{
+						&ast.Ident{Name: "hidePanics"},
+					},
+					Tok: token.ASSIGN,
+					Rhs: []ast.Expr{
+						&ast.Ident{Name: "hide"},
+					},
+				},
+			},
+		},
+	},
+}
+
+var printanyHexCase = &ast.CaseClause{
+	List: []ast.Expr{
+		&ast.Ident{Name: "hex"},
+	},
+	Body: []ast.Stmt{
+		&ast.ExprStmt{
+			X: &ast.CallExpr{
+				Fun: &ast.Ident{Name: "print"},
+				Args: []ast.Expr{
+					&ast.Ident{Name: "v"},
+				},
+			},
+		},
+	},
+}
+
+var panicprintDecl = &ast.FuncDecl{
+	Name: &ast.Ident{Name: "panicprint"},
+	Type: &ast.FuncType{Params: &ast.FieldList{
+		List: []*ast.Field{
+			&ast.Field{
+				Names: []*ast.Ident{
+					&ast.Ident{Name: "args"},
+				},
+				Type: &ast.Ellipsis{
+					Elt: &ast.InterfaceType{
+						Methods: &ast.FieldList{},
+					},
+				},
+			},
+		},
+	}},
+	Body: &ast.BlockStmt{List: []ast.Stmt{
+		&ast.IfStmt{
+			Cond: &ast.Ident{Name: "hidePanics"},
+			Body: &ast.BlockStmt{List: []ast.Stmt{
+				&ast.ReturnStmt{},
+			}},
+		},
+		&ast.RangeStmt{
+			Key:   &ast.Ident{Name: "_"},
+			Value: &ast.Ident{Name: "arg"},
+			Tok:   token.DEFINE,
+			X:     &ast.Ident{Name: "args"},
+			Body: &ast.BlockStmt{
+				List: []ast.Stmt{
+					&ast.ExprStmt{
+						X: &ast.CallExpr{
```

Comment:
join with the above (same as always - composite lit with a single composite lit field)